### DISCUSSION TITLE
Get multiple struct definitions at once

### DISCUSSION THREAD

**Discussion by michaelvantelgen-work** [Dlubal-Software/Dlubal_CSharp_Client#9] [Q&A] [answered]
Hi @jarabroz,

Could you share how we can get multiple structs all at once? The equivalent of GetNodes() for RFEM5 so to say?

I could do this but it seems slow:

            int[] usedNodeNumbers = RFEMModel.get_all_object_numbers(object_types.E_OBJECT_TYPE_NODE, 0);

            List<node> retrievedNodes = new List<node>();
            foreach (int node in usedNodeNumbers)
            {
                retrievedNodes.Add(RFEMModel.get_node(node));
            }

[edit]
I see this question is a duplicate of https://github.com/Dlubal-Software/Dlubal_CSharp_Client/issues/6
Sorry about that, still finding my way here.

[edit 2]
Same question would apply to setting multiple structs all at once - the equivalent of SetNodes() for RFEM5.

**Comment by jarabroz** [comment, marked as answer]
Hi @michaelvantelgen-Arcadis,

In the medium term, we do not plan to change it and implement get all and set_all.

In WebService, objects need to be serialized to XML, sent via TCP/IP to application, received, deserialize from XML to an object. Then some validations and checks are run. If we make "batching" of objects, it cloud lead to some speed gain, but it still needs to make all the steps described.
